telegram/base.py

```diff
@@ -74,9 +74,17 @@ def to_dict(self):
 
             value = self.__dict__[key]
             if value is not None:
-                if hasattr(value, 'to_dict'):
-                    data[key] = value.to_dict()
+                if type(value) == list:
```

Comment:
If I understand this part correctly, you added this to decode attributes that are marked as "Array of ..." in the API docs? 

In case I assume correctly, we usually have a `de_list` method on classes like `PhotoSize` ([Implementation](https://github.com/python-telegram-bot/python-telegram-bot/blob/master/telegram/photosize.py#L73) and [usage](https://github.com/python-telegram-bot/python-telegram-bot/blob/master/telegram/message.py#L176) examples) when needed. 

However, I do like your solution for this. Maybe it could be implemented in a separate PR, for all cases where it is needed.